roles/java/templates/application-agent-controller-info.xml.j2

```diff
@@ -53,7 +53,7 @@
     You can override this configuration with an environment variable 'APPDYNAMICS_AGENT_APPLICATION_NAME' or
     the system property '-Dappdynamics.agent.applicationName'  -->
 
-    <application-name>{{ application_name }}</application-name>
+    <application-name>{{ application_name if application_details_in_config else '' }}</application-name>
```

Comment:
Hey, for the app, tier and node names, there's a validation step that enforces these values here - https://github.com/Appdynamics/appdynamics-ansible/blob/master/roles/common/tasks/validate_parameters.yml#L16  . Does these conditional statements take care of that?

Reply:
Well, not sure if this validation requires any changes. Option `application_details_in_config` just controls if those params should go into agent config or not. Real use case if we use instrumentation roles - we can include java role with `application_deatils_in_config: no`  to make everything cleaner.